src/gpu/sycl/ref_pooling.hpp

```diff
@@ -53,18 +53,19 @@ struct ref_pooling_fwd_t : public sycl_gpu_primitive_t {
 
             const bool ok = is_fwd() && set_default_params() == status::success
                     && (src_md(0)->format_desc.blocking.inner_nblks == 0)
-                    && (utils::everyone_is(
-                                s8, src_md(0)->data_type, dst_md(0)->data_type)
-                            || utils::everyone_is(u8, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(f32, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(bf16, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(f16, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(s32, src_md(0)->data_type,
-                                    dst_md(0)->data_type))
+                    && (IMPLICATION(utils::one_of(bf16, src_md(0)->data_type),
+                            utils::one_of(bf16, dst_md(0)->data_type)))
+                    && f64 != src_md(0)->data_type
+                    && f64 != dst_md(0)->data_type
+                    && (IMPLICATION(bf16 == src_md(0)->data_type,
+                            bf16 == dst_md(0)->data_type))
+                    && (IMPLICATION(src_md(0)->data_type == s8,
+                            dst_md(0)->data_type != s8))
+                    && (IMPLICATION(src_md(0)->data_type == u8,
+                            dst_md(0)->data_type != u8))
+                    && (IMPLICATION(
+                            src_md(0)->data_type != dst_md(0)->data_type,
+                            desc_.prop_kind == forward_inference))
```

Comment:
Not sure I understand the logic here. You are disabling src and dst datatype to be the same except for bf16.
Having the same datatype for src and dst should be valid.

Reply:
In this [table](https://oneapi-src.github.io/oneDNN/dev_guide_pooling.html#data-type-support) src/dst datatypes could be different just when `forward_inference` is used. The bf16 dt is not combined with any other dt. Maybe I am misunderstanding something in the table. Let me know if this clarifies a little bit the questions. Thanks.